site/content/posts/2021-08-27-supporting-ad-oidc-workflows.md

```diff
@@ -0,0 +1,115 @@
+
+### Setup and Use AD with your Supervisor
+
+Pinniped Supervisor authenticates your users with the AD provider via the LDAP protocol, and then issues unique, short-lived, per-cluster tokens. Our previous blog post on [LDAP configuration]({{< ref "2021-06-02-first-ldap-release.md">}}), elaborates on the security considerations to support integration at the Pinniped Supervisor level instead of at the Concierge.
+
+To setup the AD configuration, once you have Supervisor configured with ingress [installed the Pinniped Supervisor]({{< ref "docs/howto/install-supervisor.md" >}}) and you have [configured a FederationDomain]({{< ref "docs/howto/configure-supervisor" >}}) to issue tokens for your downstream clusters, you can create an [ActiveDirectoryIdentityProvider](https://github.com/vmware-tanzu/pinniped/blob/main/generated/1.20/README.adoc#activedirectoryidentityprovider) in the same namespace as the Supervisor.
+Here’s what an example configuration looks like
+
+```yaml
+ apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+ kind: ActiveDirectoryIdentityProvider
+ metadata:
+   name: my-active-directory-idp
+   namespace: pinniped-supervisor
+ spec:
+
+   # Specify the host of the Active Directory server.
+   host: "activedirectory.example.com:636"
+
+   # Specify the name of the Kubernetes Secret that contains your Active Directory
+   # bind account credentials. This service account will be used by the
+   # Supervisor to perform LDAP user and group searches.
+   bind:
+     secretName: "active-directory-bind-account"
+
+ ---
+
+ apiVersion: v1
+ kind: Secret
+ metadata:
+   name: active-directory-bind-account
+   namespace: pinniped-supervisor
+ type: kubernetes.io/basic-auth
+ stringData:
+
+   # The dn (distinguished name) of your Active Directory bind account.
+   username: "CN=Bind User,OU=Users,DC=activedirectory,DC=example,dc=com"
+
+   # The password of your Active Directory bind account.
+   password: "YOUR_PASSWORD"
+ ```
+
+You can also customize the userSearch and groupSearch as shown in the examples in our reference documentation [here] ({{< ref “docs/howto/configure-supervisor-with-activedirectory.md" >}})
+
+Here is an example of what the ID token claims will look like:
+
+```yaml
+  aud:
+  - pinniped-cli
+  auth_time: 1630094468
+  exp: 1630094589
+  groups:
+  - Mammals@activedirectory.test.example.com
+  - Marine Mammals@activedirectory.test.example.com
+  iat: 1630094469
+  iss: https://pinniped-supervisor-clusterip.supervisor.svc.cluster.local/some/path
+  jti: 191709eb-b2fd-47e0-97f4-a06c48330c3a
+  nonce: c772c414388482163515103110cfcdfc
+  rat: 1630094468
+  sub: ldaps://activedirectory.test.example.com:636?base=DC%3Dactivedirectory%2CDC%3Dtest%2CDC%3Dexample%2CDC%3Dcom&sub=04030201-0605-0807-0910-111213141516
+  username: pinny@activedirectory.test.example.com
+ ```
```

Comment:
I don't think Pinniped users know what our ID tokens look like. Its not part of the user experience.

Maybe it would be easier to understand if we instead give them an example of what their username and group membership will look like in Kubernetes for the provided sample YAML config? A good example username would be `pinny@yourdomain.example.com` and an example group name might be `Seals@yourdomain.example.com`.

Reply:
Resolved based on review comments